RxSwift/Observables/Implementations/Materialize.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Materialize.swift
+//  Rx
+//
+//  Created by sergdort on 08/03/2017.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+public protocol EventType {
+    associatedtype E
+    func asEvent() -> Event<E>
+}
+
+extension Event : EventType {
+    public typealias E = Element
+    public func asEvent() -> Event<E> { return self }
+}
+
+class Obs<E>: ObserverType {
+    func on(_ event: Event<E>) {
+        
+    }
+}
+
+fileprivate final class MaterializeSink<Element>: Disposable, ObserverType {
```

Comment:
We probably can't use `Sink` here, since it's forwards only `Event` but we want to sent `onNext`